GVFS/GVFS.UnitTests.Mac/Service/MacServiceTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using GVFS.Platform.Mac;
+using GVFS.Service;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Mock.FileSystem;
+using Moq;
+using NUnit.Framework;
+using System.IO;
+
+namespace GVFS.UnitTests.Mac.Service
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class MacServiceTests
+    {
+        private const string GVFSServiceName = "GVFS.Service";
+        private const int ExpectedActiveUserId = 502;
+        private static readonly string ExpectedActiveRepoPath = Path.Combine("mock:", "code", "repo2");
+        private static readonly string ServiceDataLocation = Path.Combine("mock:", "registryDataFolder");
+
+        private MockFileSystem fileSystem;
+        private MockTracer tracer;
+        private Mock<MacPlatform> gvfsPlatformMock;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.tracer = new MockTracer();
+            this.fileSystem = new MockFileSystem(new MockDirectory(ServiceDataLocation, null, null));
+            this.gvfsPlatformMock = new Mock<MacPlatform>();
+            this.gvfsPlatformMock.Setup(p => p.GetCurrentUser()).Returns(ExpectedActiveUserId.ToString());
+            this.gvfsPlatformMock.Setup(p => p.GetUserIdFromLoginSessionId(ExpectedActiveUserId, It.IsAny<ITracer>())).Returns<int, ITracer>((x, y) => x.ToString());
+            this.gvfsPlatformMock.SetupGet(p => p.FileSystem).Returns(new MockPlatformFileSystem());
+            this.gvfsPlatformMock.SetupGet(p => p.Constants).Returns(new MacPlatform.MacPlatformConstants());
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            this.gvfsPlatformMock = null;
+            this.fileSystem = null;
+            this.tracer = null;
+        }
+
+        [TestCase]
+        public void ServiceStartTriggersAutoMountForCurrentUser()
+        {
+            Mock<IRepoRegistry> repoRegistry = new Mock<IRepoRegistry>(MockBehavior.Strict);
+            repoRegistry.Setup(r => r.AutoMountRepos(ExpectedActiveUserId));
+            repoRegistry.Setup(r => r.TraceStatus());
+
+            GVFSService service = new GVFSService(
+                this.tracer,
+                serviceName: null,
+                repoRegistry: repoRegistry.Object,
+                gvfsPlatform: this.gvfsPlatformMock.Object);
+
+            service.RunWithArgs(new string[] { $"--servicename={GVFSServiceName}" });
+
+            repoRegistry.VerifyAll();

```

Comment:
Will this `VerifyAll()` do anything? I don't think you've specified things should run once or anything. Try deleting the body of `GVFSService.RunWithArgs()` and see if this test breaks.

Reply:
`VerifyAll` when used with `MockBehavior.Strict` will cause the test to fail if a setup method is not called or a method is called and there isn't a setup.  That's why I like it.  It will verify that your assumptions for setup of the mock are correct.

Reply:
Thanks for clarifying!